akka-javasdk/src/main/scala/akka/javasdk/impl/serialization/JsonSerializer.scala

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.impl.serialization
+
+import java.io.IOException
+import java.lang
+import java.lang.reflect.InvocationTargetException
+import java.util
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+import akka.javasdk.JsonMigration
+import akka.javasdk.JsonSupport
+import akka.javasdk.annotations.Migration
+import akka.javasdk.annotations.TypeName
+import akka.javasdk.impl.NullSerializationException
+import akka.runtime.sdk.spi.BytesPayload
+import akka.util.ByteString
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.core.JsonProcessingException
+
+object JsonSerializer {
+  val JsonContentTypePrefix: String = "json.akka.io/"
+  private val KalixJsonContentTypePrefix: String = "json.kalix.io/"
+
+  final case class TypeHint(currenTypeHintWithVersion: String, allTypeHints: List[String])
+
+}
+
+class JsonSerializer {
```

Comment:
We will probably support other formats, such as proto. I haven't designed for that yet. Wanted to keep it as simple as possible first.